<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />




<title>Reference List</title>

<script src="site_libs/header-attrs-2.27/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/bootstrap.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/default.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>

<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->




</head>

<body>


<div class="container-fluid main-container">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html"></a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="schedule.html">Schedule</a>
</li>
<li>
  <a href="lectures.html">Lectures</a>
</li>
<li>
  <a href="labs.html">Lab Tutorials</a>
</li>
<li>
  <a href="previous.html">Previous Workshops</a>
</li>
<li>
  <a href="references.html">References</a>
</li>
<li>
  <a href="https://www.gmshiny.com/">GMShiny</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">Reference List</h1>

</div>


<p><strong>Software Packages</strong></p>
<p>Baken, E.K., M.L. Collyer, A. Kaliontzopoulou, and D.C. Adams. 2021.
geomorph v4.0 and gmShiny: enhanced analytics and a new graphical
interface for a comprehensive morphometric experience. <em>Methods in
Ecology and Evolution</em>. 12:2355–2363.</p>
<p>Collyer, M.L, and D.C. Adams. 2018. RRPP: An R package for fitting
linear models to high-dimensional data using residual randomization.
<em>Methods in Ecology and Evolution</em>. 9:1772-1779.</p>
<p><strong>General Overview of Geometric Morphometrics</strong></p>
<p>Adams, D.C., F. J. Rohlf, and D.E. Slice. 2013. A field comes of age:
Geometric morphometrics in the 21<sup>st</sup> century.
<em>Hystrix.</em> 24:7-14.</p>
<p><strong>General Analytical Theory of Methods</strong></p>
<p>Collyer, M.L., and D.C. Adams. 2024. Interrogating Random and
Systematic Measurement Error in Morphometric Data. <em>Evolutionary
Biology</em>. 51:179-207.</p>
<p>Adams, D.C. and M.L. Collyer. 2022. Consilience of methods for
phylogenetic analysis of variance. <em>Evolution</em>. 76:1406–1419.</p>
<p>Conaway, M.A., and D.C. Adams. 2022. An effect size for comparing the
strength of morphological integration across studies.
<em>Evolution</em>. 76: 2244-2259.</p>
<p>Collyer, M.L., and D.C. Adams. 2021. Phylogenetically aligned
component analysis. <em>Methods in Ecology and Evolution</em>.
12:359-372.</p>
<p>Collyer, M.L., M.A. Davis, and D.C. Adams. 2020. Making heads or
tails of combined landmark configurations in geometric morphometric
data. <em>Evolutionary Biology</em>. 47:193-205.</p>
<p>Adams, D.C., and M.L. Collyer. 2019. Comparing the strength of
modular signal, and evaluating alternative modular hypotheses, using
covariance ratio effect sizes with morphometric data.
<em>Evolution</em>. 73:2352-2367.</p>
<p>Adams, D.C., and M.L. Collyer. 2019. Phylogenetic comparative methods
and the evolution of multivariate phenotypes. <em>Annual Review of
Ecology, Evolution, and Systematics</em>. 50:405-425.</p>
<p>Adams, D.C. , and M.L. Collyer. 2018. Phylogenetic ANOVA: Group-clade
aggregation, biological challenges, and a refined permutation procedure.
<em>Evolution</em>. 72:1204-1215.</p>
<p>Adams, D.C., and M.L. Collyer. 2018. Multivariate phylogenetic
comparative methods: Evaluations, comparisons, and recommendations.
<em>Systematic Biology</em>. 67:14-31.</p>
<p>Adams, D.C. and M.L. Collyer. 2016. On the comparison of the strength
of morphological integration across morphometric datasets.
<em>Evolution</em>. 70:2623-2631.</p>
<p>Adams, D.C.. 2016. Evaluating modularity in morphometric data:
Challenges with the RV coefficient and a new test measure. <em>Methods
in Ecology and Evolution.</em> 7:565-572.</p>
<p>Davis, M.A., M.R. Douglas, M.L. Collyer, and M.E. Douglas. 2016.
Deconstructing a species-complex: geometric morphometric and molecular
analyses define species in the Western Rattlesnake (<em>Crotalus
viridis</em>). <em>PloS One</em>. 11(1):e0146166.</p>
<p>Collyer, M.L., D.J. Sekora, and D.C. Adams. 2015. A method for
analysis of phenotypic change for phenotypes described by
high-dimensional data. <em>Heredity</em>. 115: 357-365.</p>
<p>Denton, J.S.S., and D.C. Adams. 2015. A new phylogenetic test for
comparing multiple high-dimensional evolutionary rates suggests
interplay of evolutionary rates and modularity in lanternfishes
(Myctophiformes; Myctophidae). <em>Evolution</em>. 69:2425-2440.</p>
<p>Adams, D.C., and M.L. Collyer. 2015. Permutation tests for
phylogenetic comparative analyses of high-dimensional shape data: what
you shuffle matters. <em>Evolution</em>. 69:823-829.</p>
<p>Adams, D.C. 2014. A method for assessing phylogenetic least squares
models for shape and other high-dimensional multivariate data.
<em>Evolution</em>. 68:2675-2688.</p>
<p>Adams, D.C. 2014. A generalized K statistic for estimating
phylogenetic signal from shape and other high-dimensional multivariate
data. <em>Systematic Biology</em>. 63:685-697.</p>
<p>Adams, D.C. and R. Felice. 2014. Assessing phylogenetic morphological
integration and trait covariation in morphometric data using
evolutionary covariance matrices. <em>PLoS One</em>. 9(4):e94335.</p>
<p>Adams, D.C. 2014. Quantifying and comparing phylogenetic evolutionary
rates for shape and other high-dimensional phenotypic data.
<em>Systematic Biology</em>. 63:166-177.</p>
<p>Collyer, M. L., and D.C. Adams. 2013. Phenotypic trajectory analysis:
Comparison of shape change patterns in evolution and ecology.
<em>Hystrix</em>. 24:75-83.</p>
<p>Adams, D.C., and M.L. Collyer. 2009. A general framework for the
analysis of phenotypic trajectories in evolutionary studies.
<em>Evolution</em>. 63:1143-1154.</p>
<p>Adams, D. C., and M.L. Collyer. 2007. Analysis of character
divergence along environmental gradients and other covariates.
<em>Evolution</em>. 61:510-515.</p>
<p>Collyer, M. L., and D. C. Adams. 2007. Analysis of two-state
multivariate phenotypic change in ecological studies. <em>Ecology</em>.
88:683-692.</p>
<p>Adams, D. C. 1999. Methods for shape analysis of landmark data from
articulated structures. <em>Evolutionary Ecology Research.</em>
1:959-970.</p>
<p><strong>Further Statistical Theory Reading</strong></p>
<p>Commenges, D. 2003. Transformations which preserve exchangeability
and application to permutation tests. <em>Journal of Nonparametric
Statistics</em>. 15:171-185.</p>
<p>Anderson, M. J. (2001a). A new method for non-parametric multivariate
analysis of variance. <em>Austral Ecology</em>. 26, 32–46.</p>
<p>Anderson, M. J. (2001b). Permutation tests for univariate or
multivariate analysis of variance and regression. <em>Canadian Journal
of Fisheries and Aquatic Sciences, 58</em>. 626–639.</p>
<p>Gower J.C. 1966. Some distance properties of latent root and vector
methods used in multivariate analysis. <em>Biometrika</em>.
53:325–338.</p>




</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->


<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
